classes/publication/Repository.php

```diff
@@ -661,6 +686,66 @@ public function delete(Publication $publication)
         Hook::call('Publication::delete', [&$publication]);
     }
 
+    /**
+     * Given a Version Stage and a flag of whether the Version isMinor, 
+     * the publication's related data is being updated
+     *
+     * @hook 'Publication::updateVersionData::before' [[ &$newPublication, $publication ]]
+     */
+    public function updateVersionData(Publication $publication, VersionStage $versioningStage, bool $isMinor = true): Publication
+    {
+        $submission = Repo::submission()->get($publication->getData('submissionId'));
+        $nextAvailableVersionStage = Repo::submission()->getNextAvailableVersionData($submission, $versioningStage, $isMinor);
+
+        $newPublication = clone $publication;
```

Comment:
This function seems convoluted... It's called `updateVersionData`, but it returns a clone of the supplied `$publication` object rather than updating the instance the calling code provides. If there's not a specific reason for that, it would be less confusing to work on the current instance, I think.